### ISSUE TITLE
📎 Implement `lint/useAsConstAssertion` - `typescript-eslint/prefer-as-const`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#47]
### Description

[prefer-as-const](https://typescript-eslint.io/rules/prefer-as-const)

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by ematipico** [comment]
The original rule also promotes `as`, e.g. `as string`, which is not exactly `as const`. Should we:
- rename the rule to better communicate the intent of the rule?
- or have two rules?

**Comment by Conaclos** [comment]
> The original rule also promotes as, e.g. as string, which is not exactly as const.

Promote `x as string` instead of angle bracket syntax `<string> x`?

**Comment by ematipico** [comment]
I really don't know - I am a TS expert; I just read the rule and its description. Plus the correct example has:

```ts
let bar = 'bar' as string;
```

I leave the final decision to you, I just wanted to point out some inconsistencies I found

**Comment by Conaclos** [comment]
In fact, the rule only applies in the case where the literal value matches the asserted type.

**Comment by ematipico** [comment]
The name is misleading to me, though! 😆 That's what I am trying to say

**Comment by Conaclos** [comment]
> The name is misleading to me, though! 😆 That's what I am trying to say. Take in consideration that I am speaking as a newbie.

`useConstAssertion` or `useAsConstAssertion`?

**Comment by ematipico** [comment]
I love the second one
